### PR TITLE
[onert] Create TensorRegistries by using TrainableBackendContexts

### PR DESCRIPTION
**Pull request by ragmani**
This commit enables to create TensorRegistries for TrainableBackendContexts.

ONE-DCO-1.0-Signed-off-by: ragmani <ragmani0216@gmail.com>

### CODE REVIEW

**Review by SlavikMIPT — APPROVED** [review]
LGTM

File: runtime/onert/core/src/compiler/TensorRegistries.h [hunk 1]
```diff
@@ -55,6 +56,20 @@ class TensorRegistries
     }
   }
 
+  TensorRegistries(const backend::train::TrainableBackendContexts &backend_contexts, bool is_grad)
```
Comment (glistening) [inline runtime/onert/core/src/compiler/TensorRegistries.h:59]: `is_grad` determines whether `TensorRegistries` is for `grad` tensors or for normal tensors.

**Review by glistening — APPROVED** [review]
LGTM

**Review by jyoungyun — APPROVED** [review]
LGTM

**Pull request merged**